source/fundamentals/enterprise-auth.txt

```diff
@@ -280,3 +287,150 @@ mechanism:
+   
+         MongoClient mongoClient = MongoClients.create(
+             MongoClientSettings.builder()
+                 .applyToClusterSettings(builder ->
+                         builder.hosts(Arrays.asList(new ServerAddress("<hostname>", <port>))))
+                 .credential(credential)
+                 .build());
+
+.. _java-mongodb-oidc-gcp-imds:
+
+GCP IMDS
+++++++++++
+
+If your application runs on a GCP VM, or otherwise uses the
+`GCP Instance Metadata Service <https://cloud.google.com/compute/docs/metadata/querying-metadata>`__,
+you can authenticate to MongoDB by using {+driver-short+}'s built-in GCP
+support.
+
+You can specify GCP IMDS OIDC authentication on a ``MongoClientSettings`` object either by 
+using a ``MongoCredential`` object or as part of the connection string. Select the 
+:guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax.
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: mongodb-gcp-imds-connection-string
+    
+      Replace the ``<resource>`` placeholder with the percent-encoded application or
+      service that the OIDC access token is intended for.
+     
+      .. code-block:: java
+
+         MongoClient mongoClient = MongoClients.create("mongodb://<username>@<hostname>:<port>/?authMechanism=MONGODB-OIDC&authMechanismProperties=ENVIRONMENT:gcp,TOKEN_RESOURCE:<resource>");
+
+   .. tab:: MongoCredential
+      :tabid: mongodb-gcp-mongo-credential
+
+      Replace the ``<username>`` placeholder with the client ID of the GCP managed
+      identity, and replace the ``<resource>`` placeholder with the percent-encoded
+      application or service that the OIDC access token is intended for.
+      
+      .. code-block:: java
+
+         MongoCredential credential = MongoCredential.createOidcCredential("<username>")
+            .withMechanismProperty("TOKEN_RESOURCE", "<resource>")
+            .withMechanismProperty("ENVIRONMENT", "gcp");  
+   
+         MongoClient mongoClient = MongoClients.create(
+             MongoClientSettings.builder()
+                 .applyToClusterSettings(builder ->
+                         builder.hosts(Arrays.asList(new ServerAddress("<hostname>", <port>))))
+                 .credential(credential)
+                 .build());
+
+Custom Callback
++++++++++++++++
+
+On some platforms, such as Azure Functions and Azure Kubernetes Service (AKS), you
+must define a custom callback method to use OIDC to authenticate. To use a custom
+callback to retrieve an OIDC token, call the ``withMechanismProperty()`` method on
+your ``MongoCredential`` object to set the following options:
+
+- ``"ALLOWED_HOSTS"``: The list of allowed hostnames or IP addresses for MongoDB
+  The hostnames may include ``"*."`` as a leading wildcard, which allows for
+  matching subdomains. This property defaults to
+  ``["*.mongodb.net", "*.mongodb-qa.net", "*.mongodb-dev.net", "*.mongodbgov.net",
+  "localhost", "127.0.0.1", "::1"]``.
```

Comment:
S: looks like this value is cut off on the staging site and requires some horizontal scroll. Maybe adding a newline after  `"*.mongodb-dev.net" `could fix it:
```suggestion
  ``["*.mongodb.net", "*.mongodb-qa.net", "*.mongodb-dev.net", 
  "*.mongodbgov.net", "localhost", "127.0.0.1", "::1"]``.
```

Reply:
good call